### ISSUE TITLE
Improve enabling local wallet mode from cloud mode process.

### ISSUE THREAD

**Issue by clockworkgr** [bitshares/bitshares-ui#1625]
In the last month, I have walked at least 5 people through recovering access to their account after they enabled local wallet. 

It seems people mistakenly thing that enabling local wallet automatically converts the account use mode. They then download the .bin file as prompted and believe that everything is fine.

In fact, the wallet is empty with no keys in it. Next time, they unlock it and get the unlocked icon but with no access to their account.

I suggest that if a user is logged in in cloud mode and "enables" local wallet he gets the option/confirmation to import his account & keys to the newly created local wallet.

i.e. same way he picks a name and password for it , we ask him 
"Would you like to import your account 'xxxxxxx' and its keys to your local wallet? YES/NO"

If yes, we confirm cloud pass again and create the local wallet WITH account & keys included.

**Comment by startailcoon** [comment]
This sounds to me like a good plan. You've identified a user issue where there is lack of documentation and/or specific guidence to users.

May I suggest that we specify this even a little stronger with something like this: 

> Not importing your old keys will make the new local wallet a "stand-alone" wallet. You will not have access to any old assets within the newly created local wallet! 
>
> **Are you sure you did not mean to import the keys?**
>
> Import my old wallet / Create a new empy wallet

**Comment by wmbutler** [comment]
Can we recognize cases where the wallet is empty with no keys and disable the download of the .bin file or better yet, not present that as an option? I view this as a fairly serious bug as it relates to UI.

**Comment by clockworkgr** [comment]
@wmbutler well. it's both a bug and expected behaviour. Any time you create a new local wallet, it's empty (except if you're creating it as part of the process for creating a new account for the very first time).

So there's 2 aspects to this. 

1. The process described by me and @startailcoon at first to ensure the user can import his potentially existing cloud keys to the newly created local wallet in one step

2. Adding a check to the UI so that a "download backup" suggestion is not present if empty. perhaps we could replace the msg at bottom right with "This local wallet contains no keys" (possibly linked to the Restore/import settings page)

**Comment by wmbutler** [comment]
After some discussion, @startailcoon and I think that it makes sense for the "Create new local wallet" option to export the keys from the cloud account into the local bin file and force the user to download a backup. This needs UX.

![screen shot 2018-09-25 at 3 42 22 pm](https://user-images.githubusercontent.com/1254810/46041902-9c72b700-c0d9-11e8-9fc2-719df15a7b07.png)

**Comment by clockworkgr** [comment]
Since we're working on this, we should also point out to users more clearly that a local wallet != bin file.

The .bin file is the backup of ALL local wallets iirc.

**Comment by wmbutler** [comment]
@clockworkgr hrmmm. I don't think this distinction matters much to people. The bin is synonymous with the local wallet in that it's an encrypted of the data contained in the browser.

**Comment by wmbutler** [comment]
@clockworkgr if you are suggesting altering the copy, please state how the copy should be changed. This issue was primarily created to help people in the process of creating a local wallet if they previously used a cloud wallet.

The stated problem is that the process currently generates a blank wallet. The consensus seems to be that we should create a wallet that mirrors the cloud wallet effectively creating a local wallet for them. This would allow those users who stepped into bitshares with a cloud wallet to move to the more secure and flexible local wallet.

**Comment by clockworkgr** [comment]
I know what the stated problem is. I opened the issue in the first place :)

And yes it is a copy issue:

".. between browsers and computers using a backup file (.bin) which may contain one or more wallets."

But to be honest it still needs to be clearer.

Is anyone willing on working on a "LocalStorage browser" part of the settings menu? 

A sort of visual representation of the browser's local storage (in terms of table of contents rather than contents) so you can see how many Wallets are currently in the browser's cache (potentially how many accounts in each wallet)  etc. so you have an idea of what the .bin file actually contains if created?

**Comment by sschiessl-bcp** [comment]
For the backup cases:
One .bin file represents one local wallet. One local wallet may contain endless amount of keys. UI already shows the different local wallets in the Settings section, I dont think a LocalStorage browser would do anything as the user who run into this kind of problem wouldnt know what to look for either way.

I like the initial suggestion. Add a toggle "Add current account XYZ to local wallet" (with popup that explains both cases), with behavior described in initial descriptio. Also it should be enforced that the local wallet password is NOT the cloud login password.

**Comment by wmbutler** [comment]
Agreed. Question, could the user technically still use their cloud wallet login if they switched back? Seems to me that the answer is yes, but wanted to be sure.

**Comment by sschiessl-bcp** [comment]
> Agreed. Question, could the user technically still use their cloud wallet login if they switched back?

Yes, password does not get invalidated by the process.

**Comment by sschiessl-bcp** [comment]
@nikandrovalex

I'll implement this with your UX. Please let me know if you have any questions on the intended flow.

**Comment by wmbutler** [comment]
Sounds great.

**Comment by startailcoon** [comment]
@sschiessl-bcp @nikandrovalex how is work progressing on this? Do we have a UX that can be worked from, I can't see one here.

**Comment by wmbutler** [comment]
@nikandrovalex ?

**Comment by startailcoon** [comment]
Needs UX, open for claims
